tools/webdriver/webdriver/transport.py

```diff
@@ -10,6 +11,53 @@
 """Implements HTTP transport for the WebDriver wire protocol."""
 
 
+missing = object()
+
+
+class ResponseHeaders(Mapping):
+    """Read-only dictionary-like API for accessing response headers.
+
+    This class handles header case-insensitivity, and always returns all
+    headers with the same name (separated by commas). It doesn't ensure header
+    types (e.g. binary vs string).
+    """
+    def __init__(self, items):
+        self.headers_dict = {}
+        for key, value in items:
+            key = key.lower()
+            values = []
+            if key in self.headers_dict:
+                values = self.headers_dict[key]
+            values.append(value)
+            self.headers_dict[key] = values
+
+    def __getitem__(self, key):
+        """Get all headers of a certain (case-insensitive) name. If there is
+        more than one, the values are returned comma separated"""
+        values = self.headers_dict[key.lower()]
+        if len(values) == 1:
+            return values[0]
+        else:
+            return b", ".join(values)
```

Comment:
The docstring says the class doesn't ensure types, but we are assuming bytes here?

Reply:
Eugh, good catch. I was hoping to get away with not caring about this for now, but having this here will mean different Py2/Py3 behavior.

We could either drop the support for multiple header values (and return to last-one-wins), or we could return an array here (I think that would give us pretty surprising behavior however), or possibly some other idea I haven't thought of yet. Any preference?

Reply:
We could use a native string here, too. I believe the values are always native strings coming from the http library.

Reply:
SGTM, done.